pkgs/applications/window-managers/evilwm/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl,  libX11, libXext, libXrandr, libXrender,
+  xproto, xextproto, randrproto, renderproto, kbproto,  patches ? [] }:
+
+stdenv.mkDerivation rec {
+  name = "evilwm-1.1.1";
+
+  src = fetchurl {
+    url = "http://www.6809.org.uk/evilwm/${name}.tar.gz";
+    sha256 = "79589c296a5915ee0bae1d231e8912601fc794d9f0a9cacb6b648ff9a5f2602a";
+  };
+
+  buildInputs = [ libX11 libXext libXrandr libXrender
+                  xproto xextproto randrproto renderproto kbproto ];
+
+  prePatch = ''sed -i "s@/usr@$out@" Makefile'';
+
+  # Allow users set their own list of patches
+  inherit patches;
+
+  buildPhase = " make ";
```

Comment:
The default is a bit smarter about passing make flags and whatnot and may run hooks.